### PR TITLE
feat: node actions

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

The gap cursor logic didn't feel great the first time around

Still todo: Nice empty states to further encourage insertion of nodes in empty docs

## Changes

- Add a new "tiny" button style
- Make the `SlashCommandsPopover` component more generic to support new insertion option
- Moved some classes and constants into the utils file to avoid some circular dependencies
- Remove the old `NodeGapInsertionExtension`
- Remove the `LemonButtonWithDropdown` and add a linting rule on top of the deprecation

## How did you test this code?

Made sure everything still worked manually

interaction

### CODE REVIEW

File: frontend/__snapshots__/lemon-ui-lemon-button--sizes.png
Comment (benjackwhite) [inline frontend/__snapshots__/lemon-ui-lemon-button--sizes.png:1]: Not sure why these broke...

File: frontend/src/lib/lemon-ui/LemonButton/LemonButton.stories.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ export const TextOnly = (): JSX.Element => {
 }
 
 export const Sizes = (): JSX.Element => {
-    const sizes: LemonButtonProps['size'][] = ['small', 'medium', 'large']
+    const sizes: LemonButtonProps['size'][] = ['tiny' | 'small', 'medium', 'large']
```
Comment (benjackwhite) [inline frontend/src/lib/lemon-ui/LemonButton/LemonButton.stories.tsx:90]:
```suggestion
    const sizes: LemonButtonProps['size'][] = ['tiny', 'small', 'medium', 'large']
```

File: frontend/src/lib/lemon-ui/LemonButton/LemonButton.stories.tsx [hunk 2]
```diff
@@ -102,7 +102,7 @@ export const Sizes = (): JSX.Element => {
 }
 
 export const SizesIconOnly = (): JSX.Element => {
-    const sizes: LemonButtonProps['size'][] = ['small', 'medium', 'large']
+    const sizes: LemonButtonProps['size'][] = ['tiny' | 'small', 'medium', 'large']
```
Comment (benjackwhite) [inline frontend/src/lib/lemon-ui/LemonButton/LemonButton.stories.tsx:105]:
```suggestion
    const sizes: LemonButtonProps['size'][] = ['tiny', 'small', 'medium', 'large']
```

File: frontend/src/lib/lemon-ui/LemonButton/LemonButton.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ export interface LemonButtonPropsBase
     /** Like plain `disabled`, except we enforce a reason to be shown in the tooltip. */
     disabledReason?: string | null | false
     noPadding?: boolean
-    size?: 'small' | 'medium' | 'large'
+    size?: 'tiny' | 'small' | 'medium' | 'large'
```
Comment (benjackwhite) [inline frontend/src/lib/lemon-ui/LemonButton/LemonButton.tsx:66]: I wonder if `xsmall` is better? Fits better with the tailwind-esque xs, sm, md, lg etc.

File: frontend/src/scenes/notebooks/Nodes/NodeWrapper.tsx [hunk 4]
```diff
@@ -232,14 +217,30 @@ function NodeWrapper<T extends CustomNotebookNodeAttributes>(props: NodeWrapperP
...
+                                            onClick={(e) => {
+                                                e.stopPropagation()
+                                                setSlashCommandsPopoverVisible(true)
+                                            }}
+                                        />
+                                    </SlashCommandsPopover>
                                     {actions.map((x, i) => (
                                         <LemonButton
                                             key={i}
+                                            size="tiny"
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/NodeWrapper.tsx:241]: I think the <IconPlusMini> below should just be <IconPlus>

File: frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx [hunk 5]
```diff
@@ -453,18 +459,22 @@ export const SlashCommands = forwardRef<SlashCommandsRef, SlashCommandsProps>(fu
...
-): JSX.Element | null {
-    return (
-        <Popover
-            overlay={<SlashCommands ref={ref} {...props} mode="slash" />}
-            visible
-            referenceElement={props.decorationNode}
-        />
-    )
-})
+export const SlashCommandsPopover = forwardRef<SlashCommandsRef, SlashCommandsPopoverProps>(
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:462]: Something in here isn't working. With one node in the book, if I click add and pick something it replaces the node....

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: frontend/src/lib/lemon-ui/LemonButton/LemonButton.tsx (hunk 3 above)
Comment (daibhin) [reply frontend/src/lib/lemon-ui/LemonButton/LemonButton.tsx:66]: Yep 👍 Ideally we'd move everything to the xs, sm, etc... legend

**Review by benjackwhite — APPROVED** [review]
Lookin good. I pushed a small icon change but otherwise lets goooo

**Pull request merged**